File: frontend/src/backend/bidService.js
import {config, appFetch} from './appFetch';

export const bidding = (userId, productId, price, onSuccess, onErrors) => {
    appFetch('/bid',
        config('POST', {productId, price}), onSuccess, onErrors);
}
export const findBidsByUserId = ({page}, onSuccess) =>
    appFetch(`/bid/user?page=${page}`, config('GET'), onSuccess);

File: backend/src/main/java/es/udc/paproject/backend/model/services/BidService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Bid;
import es.udc.paproject.backend.model.exceptions.IncorrectBidUserException;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.MaxPriceBidException;
import es.udc.paproject.backend.model.exceptions.OutOfBidTimeException;

import java.math.BigDecimal;

public interface BidService {
    // Acción de realizar una puja de un producto --> excepción de pujar por algo fuera de plazo
    Bid bidding (Long userId, Long productId, BigDecimal maxPrice)
            throws InstanceNotFoundException, OutOfBidTimeException, MaxPriceBidException, IncorrectBidUserException;

    // Ver las pujas que ha realizado un usuario
    Block<Bid> findBidsByUserId (Long id, int page, int size) throws InstanceNotFoundException;


}
